src/nebari/schema.py

```diff
@@ -28,6 +28,7 @@ class Config:
         extra = "forbid"
         validate_assignment = True
         allow_population_by_field_name = True
+        smart_union = True
```

Comment:
This is the default behavior in Pydantic V2 and prevents some unintuitive behavior.  This is necessary for the new cert classes when loadiing in the config file to make sure pydantic loads the correct class.